### PR TITLE
feat: update GMA to 0.2.35

### PR DESCRIPTION
**Pull request by jplaisted**
Required editing the search tests for the new search config tests.

## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by mars-lan** [comment]
LGTM :)

File: gms/impl/src/main/resources/index/data-process/settings.json [hunk 1]
```diff
@@ -44,6 +44,13 @@
           "type": "custom",
           "tokenizer": "slash_tokenizer"
         },
+        "lowercase_keyword": {
```
Comment (camelliazhang) [inline gms/impl/src/main/resources/index/data-process/settings.json:47]: Why this change?

**Review by camelliazhang — APPROVED** [review]
Minor question, otherwise, LGTM!

File: gms/impl/src/main/resources/index/data-process/settings.json (hunk 1 above)
Comment (jplaisted) [reply gms/impl/src/main/resources/index/data-process/settings.json:47]: Good question! The test caught something :) 

I'm not sure if it was the _right_ fix, but it was a fix. The auto complete template for data process was referencing this (my guess is someone copied it from another index, but didn't copy the settings). Elasticsearch threw an error since it didn't exist.

https://github.com/linkedin/datahub/blob/d89fa4a27bdca3406e830720f650f75da6f2bdab/gms/impl/src/main/resources/dataProcessESAutocompleteQueryTemplate.json

Comment (jplaisted) [reply gms/impl/src/main/resources/index/data-process/settings.json:47]: (I figured this fix was fine to autocomplete on lower case sensitivity, but this can always be changed if that was incorrect)
